Formatter adds unnecessary empty line before (first) method comment

astares opened the pull request:
- introduce an additional setting #numberOfNewLinesAfterMethodSignatureWithMethodComment beside existing #numberOfNewLinesAfterMethodSignature

Fix #12175

astares committed:
Formatter adds unnecessary empty line before (first) method comment 

 - introduce an additional setting #numberOfNewLinesAfterMethodSignatureWithMethodComment beside existing #numberOfNewLinesAfterMethodSignature

Fix #12175

astares commented:
We can now distinguish in the settings:

![image](https://user-images.githubusercontent.com/5980033/210119495-fe11d425-ce90-44e2-9b1f-ce63168b6385.png)

- between methods with a comment (where we want to only have 1 newline to directly follow in the next line after method signature) 
- and methods without a comment (where we want to only have 2 newlines to have an empty line after the method signature before the method body starts)

astares commented:
Tests are green in image:

![image](https://user-images.githubusercontent.com/5980033/210119765-7c6a2e47-f093-42d6-867b-6720618cc8f5.png)

Lets see what the CI tells us.

Ducasse commented:
Tx. I'm always annoyed by the extra line before the first comment. I should have fixed it long time ago.

gcotelli commented on src/EnlumineurFormatter/EFContext.class.st:
```diff
@@ -287,6 +288,17 @@ EFContext >> numberOfNewLinesAfterMethodSignature: aBoolean [
+]
+
+{ #category : #accessing }
+EFContext >> numberOfNewLinesAfterMethodSignatureWithMethodComment: aBoolean [
+	newLinesAfterMethodSignatureWithMethodComment := aBoolean
```
```suggestion
EFContext >> numberOfNewLinesAfterMethodSignatureWithMethodComment: anInteger [
	newLinesAfterMethodSignatureWithMethodComment := anInteger
```

astares replied:
@gcotelli Good catch!
But then this is already wrong in the setting that was there before (newLinesAfterMethodSignature) which I just used as template.

I will change it

astares replied:
@gcotelli fixed in https://github.com/pharo-project/pharo/pull/12178/commits/00831674cee35ea901bb14f75fd17f03c83a2fda

astares committed:
Merge branch 'pharo-project:Pharo11' into 12175-Formatter-adds-unnecessary-empty-line-before-first-method-comment

astares committed:
Use anInteger and not aBoolean in
#numberOfNewLinesAfterMethodSignatureWithMethodComment:
#numberOfNewLinesAfterMethodSignature:
#numberOfNewLinesAfterMethodComment:

Also used anInteger instead of aNumber in 
#numberOfSpacesAfterCaretSymbolInReturn:
#numberOfSpacesInsideBlock:
#numberOfSpacesInsideParentheses:
#numberOfSpacesAfterCaretSymbolInReturn:

Also use aBoolean instead of anInteger in 
#newLineBeforeFirstCascade:
#newLineAfterCascade:

astares commented:
@gcotelli / @MarcusDenker Thanks!
